cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/NativeDependencyResolverImpl.java

```diff
@@ -0,0 +1,312 @@
+        if (dependencies != null) {
+            modulePlatformLibs.addAll(dependencies);
+        }
+    }
+
+    private void addPlatformLibs(PackageID packageID, List<Path> modulePlatformLibs) {
+        List<Path> platformLibs = new ArrayList<>();
+        List<Library> libraries = manifest.getPlatform().libraries;
+
+        Optional<Dependency> importPathDependency = buildContext.getImportPathDependency(packageID);
+        if (libraries != null) {
+            for (Library library : libraries) {
+                if ((library.getModules() == null ||
+                        Arrays.asList(library.getModules()).contains(packageID.name.value) ||
+                        Arrays.asList(library.getModules()).contains(packageID.orgName.value + "/" +
+                                packageID.name.value)) &&
+                        !(library.getScope() != null && library.getScope().equalsIgnoreCase("testOnly"))) {
+                    String libFilePath = library.getPath();
+                    if (libFilePath == null) {
+                        continue;
+                    }
+
+                    Path nativeFile = sourceRootPath.resolve(Paths.get(libFilePath));
+                    if (importPathDependency.isPresent()) {
+                        platformLibs.add(nativeFile.getFileName());
+                    }
+                    modulePlatformLibs.add(nativeFile);
+                }
+            }
+        }
+
+        importPathDependency.ifPresent(dependency -> validateBaloDependencies(packageID, platformLibs,
+                dependency.getMetadata().getPath()));
+    }
+
+    private void validateBaloDependencies(PackageID packageID, List<Path> platformLibs, Path importDependencyPath) {
+        Manifest manifestFromBalo = RepoUtils.getManifestFromBalo(importDependencyPath);
+        List<Library> baloDependencies = manifestFromBalo.getPlatform().libraries;
+        List<Path> baloCompileScopeDependencies = new ArrayList<>();
+        if (baloDependencies == null) {
+            return;
+        }
+
+        for (Library baloTomlLib : baloDependencies) {
+            if (baloTomlLib.getScope() != null && baloTomlLib.getScope().equalsIgnoreCase("provided")) {
+                baloCompileScopeDependencies.add(Paths.get(baloTomlLib.getPath()).getFileName());
+            }
+        }
+
+        for (Path baloTomlLib : baloCompileScopeDependencies) {
+            if (!platformLibs.contains(baloTomlLib)) {
+                buildContext.out().println("warning: " + packageID + " is missing a native library dependency - " +
+                        baloTomlLib);
+            }
+        }
+    }
+
+    private void addLibsFromBalo(Path baloFilePath, List<Path> moduleDependencySet) {
+
+        String fileName = baloFilePath.getFileName().toString();
+        Path baloFileUnzipDirectory = Paths.get(baloFilePath.getParent().toString(),
+                fileName.substring(0, fileName.lastIndexOf(".")));
```

Comment:
```suggestion
                fileName.substring(0, fileName.lastIndexOf('.')));
```
Can we make this a `char` argument?